addons/mail/static/src/messaging/entity_field.js

```diff
@@ -428,9 +428,16 @@ class EntityField {
             const newVal = [];
             for (const otherEntity of entity[relationName]) {
                 const otherField = otherEntity.constructor.fields[relatedFieldName];
-                const v = otherField.get(otherEntity);
-                if (v) {
-                    newVal.push(v);
+                const otherValue = otherField.get(otherEntity);
+                if (otherValue) {
+                    if (otherValue instanceof Array) {
```

Comment:
should check definition of related field, that it's a relational field of type `x2many`. If this is an attribute containing an array, we definitely want to keep nested arrays.

Reply:
It is needed for arrays too, because when targeting with a related field a field that is also a related of `attr` through a o2m it must be flattened (but the type of the middle related  is `attr`).

I think we should not support any type of field that would be an array, this is indicating a relation should be made instead. We should also not support object fields as most comparison/equality logic won't work as expected.

If we really want to support it, this condition should become more complex, involving checking recursively if the target field can be a related of a x2many at some point.

Reply:
task-2261221